pkg/network/node/networkpolicy.go

```diff
@@ -263,6 +263,42 @@ func (np *networkPolicyPlugin) SyncVNIDRules() {
 	}
 }
 
+func (np *networkPolicyPlugin) selectPodsFromNamespaces(nsLabelSel, podLabelSel *metav1.LabelSelector) map[uint32]string {
+	namespaces := make(map[string]uint32)
+	vnidsIPs := make(map[uint32]string)
+
+	nsSel, err := metav1.LabelSelectorAsSelector(nsLabelSel)
+	if err != nil {
+		// Shouldn't happen
+		utilruntime.HandleError(fmt.Errorf("ValidateNetworkPolicy() failure! Invalid NamespaceSelector: %v", err))
+		return nil
+	}
+
+	podSel, err := metav1.LabelSelectorAsSelector(podLabelSel)
+	if err != nil {
+		// Shouldn't happen
+		utilruntime.HandleError(fmt.Errorf("ValidateNetworkPolicy() failure! Invalid PodSelector: %v", err))
+		return nil
+	}
+
+	for vnid, ns := range np.namespaces {
+		if kns, exists := np.kNamespaces[ns.name]; exists {
+			if nsSel.Matches(labels.Set(kns.Labels)) {
+				namespaces[ns.name] = vnid
+			}
+		}
+	}
+	for _, pod := range np.pods {
+		vnid, exists := namespaces[pod.Namespace]
+		if exists && podSel.Matches(labels.Set(pod.Labels)) {
+			vnidsIPs[vnid] = pod.Status.PodIP
```

Comment:
This will only work for a single matching pod per namespace. I think you need to return an array of {vnid, podIP}. (Or else just do the fmt.Sprintf here and return an array of strings; in that case you should probably change selectPods and selectNamespaces to match, for consistency.)